components/ws-manager/pkg/manager/status.go

```diff
@@ -354,11 +354,6 @@ func (m *Manager) extractStatusFromPod(result *api.WorkspaceStatus, wso workspac
 			// While the pod is being deleted we do not care or want to know about any failure state.
 			// If the pod got stopped because it failed we will have sent out a Stopping status with a "failure"
 			result.Conditions.Failed = ""
-		} else {
-			if _, ok := pod.Annotations[workspaceNeverReadyAnnotation]; ok {
```

Comment:
why this part is being removed?

Reply:
Because I think if the workspace pod is **never ready**, it still needs to go through the [Stopping](https://github.com/gitpod-io/gitpod/blob/093e8c6586183660a4bdb9f218e4afd12ca1ad9e/components/ws-manager/pkg/manager/monitor.go#L430) phase to do extra logic (e.g.: resource clean up). The idea is like the Kubernetes controller reconcile loop, and we do extra logic if the resource is Terminating with the finalizer protected, and then remove the finalizer.

From my testing, the scenario ErrImagePull makes the `status.Conditions.Failed != ""`, so the annotation `workspaceFailedBeforeStoppingAnnotation` be added.
https://github.com/gitpod-io/gitpod/blob/093e8c6586183660a4bdb9f218e4afd12ca1ad9e/components/ws-manager/pkg/manager/monitor.go#L333-L341

During the workspace pod in the Stopping state, it contains the annotation `workspaceFailedBeforeStoppingAnnotation` and `workspaceNeverReadyAnnotation`.
https://github.com/gitpod-io/gitpod/blob/093e8c6586183660a4bdb9f218e4afd12ca1ad9e/components/ws-manager/pkg/manager/monitor.go#L487-L501

Reply:
@utam0k
Do you know how to simulate the workspacekit faiI?
I want to check for this PR changes, would it break your previous contribution #10085.

Reply:
Ah, it makes sense then.
Only concern is to do some extra testing, just to make sure there is no code that was made on the assumption of this code existing. 
I think you already tested regular workspaces, can you also test prebuilds and image builds? Just to be extra safe.

Reply:
@jenting 
- Make this time short
     https://github.com/gitpod-io/gitpod/blob/90b8032d1c78e8d21549dbd5a01eacf7d056d860/components/workspacekit/cmd/rings.go#L971
- Change this socket file name
    https://github.com/gitpod-io/gitpod/blob/90b8032d1c78e8d21549dbd5a01eacf7d056d860/components/ws-daemon/pkg/iws/iws.go#L156

Reply:
Thank you @sagor999 for the suggestion and @utam0k for the hack way.

- prebuild
<img width="830" alt="image" src="https://user-images.githubusercontent.com/49380831/181515337-ffe0609b-117d-41fa-97ac-8f5dbd3be26f.png">
<img width="340" alt="image" src="https://user-images.githubusercontent.com/49380831/181515386-98ab8280-b26a-42b9-9afc-69cb65f35688.png">
- image build
<img width="934" alt="image" src="https://user-images.githubusercontent.com/49380831/181515809-72a1a844-a51f-4c36-9825-23b1e5554afc.png">
<img width="864" alt="image" src="https://user-images.githubusercontent.com/49380831/181515873-332eba28-2dd2-497f-8ee9-88f3fe2bd168.png">

Reply:
I am confused. Why those screenshots show errors? 
I would have expected that prebuilds\imagebuild work with this change? Or am I misreading something here?

Reply:
Or is this testing workspacekit failure?

Reply:
> Or is this testing workspacekit failure? 

Sport for missing add the context.

Yes, I was testing the workspacekit failure against regular, prebuild, and imagebuild cases, verify that the workspace pod is able to be terminated and removed, and also the pod phased is stopped in the end.